src/main/java/org/openelisglobal/common/rest/provider/NonConformEventsRestController.java

```diff
@@ -0,0 +1,250 @@
+package org.openelisglobal.common.rest.provider;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import org.openelisglobal.common.exception.LIMSInvalidConfigurationException;
+import org.openelisglobal.common.provider.query.PatientSearchResults;
+import org.openelisglobal.common.services.DisplayListService;
+import org.openelisglobal.common.services.RequesterService;
+import org.openelisglobal.common.util.DateUtil;
+import org.openelisglobal.login.valueholder.UserSessionData;
+import org.openelisglobal.qaevent.form.NonConformingEventForm;
+import org.openelisglobal.qaevent.service.NceCategoryService;
+import org.openelisglobal.qaevent.valueholder.NcEvent;
+import org.openelisglobal.qaevent.worker.NonConformingEventWorker;
+import org.openelisglobal.sample.service.SampleService;
+import org.openelisglobal.sample.valueholder.Sample;
+import org.openelisglobal.sampleitem.service.SampleItemService;
+import org.openelisglobal.sampleitem.valueholder.SampleItem;
+import org.openelisglobal.search.service.SearchResultsService;
+import org.openelisglobal.systemuser.service.SystemUserService;
+import org.openelisglobal.systemuser.valueholder.SystemUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class NonConformEventsRestController {
+
+  private final SampleService sampleService;
+  private final SampleItemService sampleItemService;
+  private final SearchResultsService searchResultsService;
+  private final NonConformingEventWorker nonConformingEventWorker;
+  private final NceCategoryService nceCategoryService;
+  private final RequesterService requesterService;
+
+  private static final String USER_SESSION_DATA = "userSessionData";
+
+  @Autowired
+  private SystemUserService systemUserService;
+
+  public NonConformEventsRestController(
+    SampleService sampleService,
+    SampleItemService sampleItemService,
+    SearchResultsService searchResultsService,
+    NonConformingEventWorker nonConformingEventWorker,
+    NceCategoryService nceCategoryService,
+    RequesterService requesterService
+  ) {
+    this.sampleService = sampleService;
```

Comment:
@konavivekramakrishna i dont see exactly where you innitialize these services in this controller